src/mpi/romio/adio/ad_lustre/ad_lustre_wrcoll.c

```diff
@@ -576,7 +570,9 @@ static void ADIOI_LUSTRE_Exch_and_write(ADIO_File fd, const void *buf,
                         recv_count[i]++;
                         ADIOI_Assert((((ADIO_Offset) (uintptr_t) write_buf) + req_off - off) ==
```

Comment:
I think this line of assertion can be removed, because your change of
using MPI_Get_address and `addr` is of type MPI_Aint.